functional-tests/tests/rpc_recent_blocks.py

```diff
@@ -18,14 +17,11 @@ def main(self, ctx: flexitest.RunContext):
 
         # create both btc and sequencer RPC
         seqrpc = seq.create_rpc()
-        counter = 0
-        while counter <= 20:
-            blk = seqrpc.strata_getHeadersAtIdx(NUM_BLOCKS_TO_RECEIVE)
-            if blk is None:
-                counter += 1
-                time.sleep(1)
-            else:
-                break
+        wait_until(
+            lambda: seqrpc.strata_getHeadersAtIdx(NUM_BLOCKS_TO_RECEIVE) is not None,
+            error_with=f"Expected block {NUM_BLOCKS_TO_RECEIVE} not generated",
+            timeout=20,
+        )
```

Comment:
There's some slight nuance here because the constant is phrased as if it's expecting to get n blocks, but actually we're just checking to see if the nth block has been generated.  If this test runs later in the sequence and we're on block 100, then this won't be checking what we expect it to check, which I would assume to be "block 100+10 has been generated".